### PR TITLE
Removed Gateway Virtual Node and replaced it with Virtual Gateway.

### PR DESCRIPTION
**Pull request by sshver**
*Issue #, if available:* None

*Description of changes:*

Updated `cross-account` walkthrough to remove gateway image (NGINX proxy which listens on a given port and forwards requests to the backend service) and replaced it with a Virtual Gateway.

Pending:
- [x] Update ReadMe

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: walkthroughs/howto-cross-account/vars.env [hunk 1]
```diff
@@ -2,12 +2,12 @@ unset AWS_ACCESS_KEY_ID
 unset AWS_SECRET_ACCESS_KEY
 export AWS_PRIMARY_ACCOUNT_ID=
 export AWS_SECONDARY_ACCOUNT_ID=
-export AWS_PRIMARY_PROFILE=primary
-export AWS_SECONDARY_PROFILE=secondary
```
Comment (rajal-amzn) [inline walkthroughs/howto-cross-account/vars.env:6]: nit: I guess these defaults are fine to leave as-is.

**Review by rajal-amzn — APPROVED** [review]

File: walkthroughs/howto-cross-account/primary-account/app.yaml [hunk 2]
```diff
@@ -228,15 +199,14 @@ Resources:
             awslogs-stream-prefix: 'gateway-envoy'
         Environment:
         - Name: 'APPMESH_VIRTUAL_NODE_NAME'
```
Comment (dfezzie) [inline walkthroughs/howto-cross-account/primary-account/app.yaml:201]: We should update this to `APPMESH_RESOURCE_ARN` as specified in: https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html

File: walkthroughs/howto-cross-account/primary-account/mesh.yaml [hunk 3]
```diff
@@ -9,22 +9,34 @@ Resources:
...
+    Type: AWS::AppMesh::GatewayRoute
+    Properties: 
+      GatewayRouteName: gateway-gr
+      MeshName: !GetAtt Mesh.MeshName
+      Spec:
+        HttpRoute:
+          Action:
+            Target:
+              VirtualService:
+                VirtualServiceName: !Sub 'backend.${ProjectName}.local'
```
Comment (dfezzie) [inline walkthroughs/howto-cross-account/primary-account/mesh.yaml:36]: This could be replaced with `!GetAtt VirtualService.VirtualServiceName`

File: walkthroughs/howto-cross-account/vars.env [hunk 4]
```diff
@@ -4,10 +4,10 @@ export AWS_PRIMARY_ACCOUNT_ID=
...
-export PROJECT_NAME=cross-account
-export AWS_DEFAULT_REGION=us-west-2
+export PROJECT_NAME=
+export AWS_DEFAULT_REGION=
 export KEY_PAIR=
 
-export ENVOY_IMAGE=840364872350.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/aws-appmesh-envoy:v1.12.2.1-prod
+export ENVOY_IMAGE=840364872350.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/aws-appmesh-envoy:v1.15.1.0-prod
 export BACKEND_1_IMAGE=dockercloud/hello-world
 export BACKEND_2_IMAGE=karthequian/helloworld
```
Comment (dfezzie) [inline walkthroughs/howto-cross-account/vars.env:13]: Why do we have these vars to begin with? None of our other projects use this method (AFAIK). Do we not instruct the user to populate these in the README?

File: walkthroughs/howto-cross-account/secondary-account/deploy.sh [hunk 5]
```diff
@@ -86,6 +86,8 @@ delete_cfn_stack() {
 }
 
 delete_stacks() {
+    echo "Deleting route to cleanly delete ${PROJECT_NAME}-mesh"
+    aws --profile ${AWS_PRIMARY_PROFILE} appmesh delete-route --mesh-name ${PROJECT_NAME}-mesh --virtual-router-name backend-vr --route-name backend-route
```
Comment (dfezzie) [inline walkthroughs/howto-cross-account/secondary-account/deploy.sh:90]: Why is this required?

File: walkthroughs/howto-cross-account/README.md [hunk 6]
```diff
@@ -6,7 +6,7 @@ This example shows how to enable App Mesh between multiple accounts for cross ac
 
 ### Gateway
 
-The gateway image is an NGINX proxy which listens on a given port and forwards requests to the backend service on a given port. It also responds to NLB HealthChecks.
+Virtual gateway allows resources that are outside of your mesh to communicate to resources that are inside of your mesh. Gateway Routes specify the routing rules for inbound traffic to virtual services within your mesh
```
Comment (dfezzie) [inline walkthroughs/howto-cross-account/README.md:9]: nit: `Virtual gateways allow resources...`

**Review by dfezzie — COMMENTED** [review]
Couple questions on changes that were made

File: walkthroughs/howto-cross-account/vars.env (hunk 4 above)
Comment (sshver) [reply walkthroughs/howto-cross-account/vars.env:13]: Our other walkthroughs do not have this `vars.env` file. But in this walkthrough, the README has instructions to update the `vars.env` and add the appropriate account and profile settings.

File: walkthroughs/howto-cross-account/secondary-account/deploy.sh (hunk 5 above)
Comment (sshver) [reply walkthroughs/howto-cross-account/secondary-account/deploy.sh:90]: On deleting the mesh stack, I was getting `ResourceInUse` exception. This is because we are updating the route (resource of primary account) to hit the application deployed in the secondary account. So I am deleting the route in the mesh of the primary account before deleting the mesh in the secondary account.

File: walkthroughs/howto-cross-account/vars.env (hunk 4 above)
Comment (dfezzie) [reply walkthroughs/howto-cross-account/vars.env:13]: I don't love this, but I won't block on it since we are not concerned about updating this walk through with this PR

**Review by dfezzie — APPROVED** [review]
